src/plugins/discover/public/application/main/components/top_nav/app_menu_actions/get_alerts.tsx

```diff
@@ -0,0 +1,173 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the "Elastic License
+ * 2.0", the "GNU Affero General Public License v3.0 only", and the "Server Side
+ * Public License v 1"; you may not use this file except in compliance with, at
+ * your election, the "Elastic License 2.0", the "GNU Affero General Public
+ * License v3.0 only", or the "Server Side Public License, v 1".
+ */
+
+import React, { useCallback, useMemo } from 'react';
+import type { DataView } from '@kbn/data-plugin/common';
+import { i18n } from '@kbn/i18n';
+import {
+  AppMenuActionId,
+  AppMenuActionSubmenuSecondary,
+  AppMenuActionType,
+} from '@kbn/discover-utils';
+import {
+  AlertConsumers,
+  ES_QUERY_ID,
+  RuleCreationValidConsumer,
+  STACK_ALERTS_FEATURE_ID,
+} from '@kbn/rule-data-utils';
+import { RuleTypeMetaData } from '@kbn/alerting-plugin/common';
+import { DiscoverStateContainer } from '../../../state_management/discover_state';
+import { AppMenuDiscoverParams } from './types';
+
+const EsQueryValidConsumer: RuleCreationValidConsumer[] = [
+  AlertConsumers.INFRASTRUCTURE,
+  AlertConsumers.LOGS,
+  AlertConsumers.OBSERVABILITY,
+  STACK_ALERTS_FEATURE_ID,
+];
+
+interface EsQueryAlertMetaData extends RuleTypeMetaData {
+  isManagementPage?: boolean;
+  adHocDataViewList: DataView[];
+}
+
+const CreateAlertFlyout: React.FC<{
+  getDiscoverParams: () => AppMenuDiscoverParams;
+  onFinishAction: () => void;
+  stateContainer: DiscoverStateContainer;
+}> = React.memo(({ stateContainer, getDiscoverParams, onFinishAction }) => {
```

Comment:
Curious why we're memoizing this component?

Reply:
Removed